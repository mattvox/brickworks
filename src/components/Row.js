import applyCssWithUnit from '../utils/applyCssWithUnit';
import Grid from './Grid';

export default Grid.extend`
  flex: 0 1 100%;
  flex-wrap: ${({ wrap }) => wrap || 'nowrap'};
  flex-direction: ${({ direction }) => direction || 'inherit'};
  justify-content: ${({ justifyContent }) => justifyContent || 'inherit'};
  align-content: ${({ alignContent }) => alignContent || 'inherit'};
  align-items: ${({ alignItems }) => alignItems || 'inherit'};
  padding: ${({ padding }) => (padding && applyCssWithUnit(padding)) || 0};
  margin: ${({ margin }) => (margin && applyCssWithUnit(margin)) || 0};
`;
