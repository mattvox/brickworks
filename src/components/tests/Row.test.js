import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';

import Row from '../Row';

Enzyme.configure({ adapter: new Adapter() });

describe('<Row />', () => {
  const wrapper = shallow(<Row />);

  it('should render correctly', () => {
    expect(wrapper.find('div').exists()).toBe(true);
  });

  it('should match snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should have default style - flex: 0 1 100%', () => {
    expect(wrapper.find('div')).toHaveStyleRule('flex', '0 1 100%');
  });

  it('should have default style - flex-flow: row wrap', () => {
    expect(wrapper.find('div')).toHaveStyleRule('flex-flow', 'row wrap');
  });

  it('should have default style - justify-content: center', () => {
    expect(wrapper.find('div')).toHaveStyleRule('justify-content', 'center');
  });

  it('should have default style - order: 0', () => {
    expect(wrapper.find('div')).toHaveStyleRule('order', '0');
  });
});
