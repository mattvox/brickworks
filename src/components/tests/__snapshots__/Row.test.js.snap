// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 0 1 100%;
  -ms-flex: 0 1 100%;
  flex: 0 1 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: inherit;
  -ms-flex-direction: inherit;
  flex-direction: inherit;
  -webkit-box-pack: inherit;
  -webkit-justify-content: inherit;
  -ms-flex-pack: inherit;
  justify-content: inherit;
  -webkit-align-content: inherit;
  -ms-flex-line-pack: inherit;
  align-content: inherit;
  -webkit-align-items: inherit;
  -webkit-box-align: inherit;
  -ms-flex-align: inherit;
  align-items: inherit;
}

<div
  className="c0"
/>
`;
